cli/pcluster/config_sanity.py

```diff
@@ -25,424 +25,457 @@
+                    mt_id = item.get("MountTargetId")
+            # If there is an existing mt in the az, need to check the inbound and outbound rules of the security groups
+            if mt_id:
+                nfs_access = False
+                in_access = False
+                out_access = False
+                # Get list of security group IDs of the mount target
+                sg_ids = efs.describe_mount_target_security_groups(MountTargetId=mt_id).get("SecurityGroups")
+                for sg in ec2.describe_security_groups(GroupIds=sg_ids).get("SecurityGroups"):
+                    # Check all inbound rules
+                    in_rules = sg.get("IpPermissions")
+                    for rule in in_rules:
+                        if self.__check_sg_rules_for_port(rule, 2049):
+                            in_access = True
+                            break
+                    out_rules = sg.get("IpPermissionsEgress")
+                    for rule in out_rules:
+                        if self.__check_sg_rules_for_port(rule, 2049):
+                            out_access = True
+                            break
+                    if in_access and out_access:
+                        nfs_access = True
+                        break
+                if not nfs_access:
+                    self.__fail(
+                        "EFSFSId",
+                        "There is an existing Mount Target %s in the Availability Zone %s for EFS %s, "
+                        "and it does not have a security group with inbound and outbound rules that support NFS. "
+                        "Please modify the Mount Target's security group, or delete the Mount Target."
+                        % (mt_id, availability_zone, resource_value[0]),
+                    )
+
+                return True
         except ClientError as e:
-            _fail(resource_type, e.response.get("Error").get("Message"))
-    if resource_type == "EC2IAMRoleName":
-        try:
-            iam = boto3.client(
-                "iam",
-                region_name=region,
-                aws_access_key_id=aws_access_key_id,
-                aws_secret_access_key=aws_secret_access_key,
-            )
+            self.__fail("EFSFSId", e.response.get("Error").get("Message"))
 
-            arn = iam.get_role(RoleName=resource_value).get("Role").get("Arn")
-            accountid = (
-                boto3.client(
-                    "sts",
-                    region_name=region,
-                    aws_access_key_id=aws_access_key_id,
-                    aws_secret_access_key=aws_secret_access_key,
+    def validate(self, resource_type, resource_value):  # noqa: C901 FIXME
+        """
+        Validate the given resource. Print an error and exit in case of error.
+
+        :param resource_type: Resource type
+        :param resource_value: Resource value
+        :return True or False (for the EFSFSId resource type only), nothing in the other cases
```

Comment:
This sounds really weird. Is there any way to fix it and to have the function either return always something or never?

Reply:
Good catch. Fixed.